TeachEasy/src/teacheasy/data/ButtonData.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Sam Hall and Sam Raeburn
+ * 
+ * Copyright (c) 2015 Sofia Software Solutions. All Rights Reserved.
+ * 
+ */
+package teacheasy.data;
+/** Class to hold Button Data 
+ * @version 1.0 05 Feb 2015
+ * @author sh1157 and sr896
+ * */
+public class ButtonData extends PageObject {
+	
+	public static enum ButtonInteractionType {
+		NORMAL
+	}
+	
+	private float xEnd, yEnd;
+	private float outlineThickness;
+	private boolean outline;
+	private boolean shadow;
+	private boolean visible;
+	private String text;
+	private int function;
+	private ButtonInteractionType interaction;
+	
+	/** Contructor for ButtonData */
+	public ButtonData(PageObjectType nType, float nXStart, float nYStart) {
+		super(nType, nXStart, nYStart);

```

Comment:
The constructor should initialise values for all the variables. If it does not there is a chance that a get function will be called before the associated set function, and a null pointer exception will occur.

Hence parameters should be created for the constructor for each of the variables maintained in this class.

For an example of the best way to do this check out calum and lewis's work on the GraphicsObject branch.